### DISCUSSION TITLE
RGB Lighting for Beginners or performance/live mode

### DISCUSSION THREAD

**Discussion by brassmonk** [FundamentalFrequency/.github#7] [Ideas]
I wanted to see if there is the possibility of adding in RGB to the board along with control in software to perhaps provide visual aids like showing chords or perhaps aesthetically in a performance/live mode?

**Comment by pixiesue** [comment]
Would it be useful to solder RGB keyswitches to the PCB? To futureproof our LMN builds.

**Reply by HowdyHolto** [reply]
I'm too am curious about RGB capability on maybe a later rev of the board. I'm thinking of it from a software/hardware UX and UI perspective. Using light and vibration to guide the user and make operation even more intuitive. With RGB, we can change the functionality over time since it's opensource and not back ourselves into a corner. Same on the Encoders. I've been looking for hardware that matches some of this, but I'm pretty novice on the hardware side of things. 

I'm sure as more and more experienced hardware folks get wind of it they will also chip in to help Stone out on iterations. Right now I'm thinking he's likely getting inundated with all manner of requests. ha.

**Reply by pixiesue** [reply]
Thank you for your answer, i know nothing about anything useful to the project right now.
Obviously we all wish him well :) Just talking about it is kind of contributing as a show of interest at that point 
(github feels intimidating for "low level" contributions though).

Do you mean the actual PCB would not support hypothetical 5 pin rgb keyboard switches ?
But it would be very useful to have rgb led capable keyswitches as a synth user at some point?

Should I order RGB keyswitches ? I already want to use a few black keys and maybe colour encoders ?
Any ideas to improve our keys choice is more than welcome of course.

![image](https://user-images.githubusercontent.com/107801377/174499650-7b6d0b06-1864-4d43-ba5c-3a2a49ec5420.png)

**Reply by HowdyHolto** [reply]
Stone answered right below me earlier. Sounds like he's open to adding support! **I'd wait until he confirms, however before ordering switches.** 

I feel you on GitHub. I've been around it a lot but only as someone submitting requests at my day job. I'm excited to contribute as well and looking forward to seeing all the community interest. This is such a cool project! He's done a hell of a lot of work.

**Reply by stonepreston** [reply]
@pixiesue, im not super into the mechanical keyboard community, but from what I understand the switches themselves usually dont have LEDs on them. They just have slots for you to insert an LED which is then soldered to the board along with the switch. This means that the PCB would have to be modified to accept per-key LEDs. You wont be able to use LEDs on the board until that happens.

**Reply by stonepreston** [reply]
@pixiesue Regarding the encoders, the color coding is great, and I think it will be easiest to just design some stickers that can be printed on a color laser printer much like the keycap icons. This way things are flexible for many different encoders, and modifying them could be as simple as scaling the SVG to fit your encoder. 

same with the recording button, its probably best to just make the record icon red and print that out. 

the black keys are nice, but I think having the keys a uniform color looks a lot cleaner

**Comment by stonepreston** [comment]
Definitely open to adding rgb to the board. Shouldn’t be an issue

**Reply by pixiesue** [reply]
Thank you for your quick response, I was about to order PCBs relatively soon, should we wait for a revision ?

I made a thread about an European group buy, would @jantuitman be interested for that upgrade ?
I am willing to help you build a LMN3 if you wish, i guess you would handle the Pi4/Screen part of the build yourself.

**Reply by jantuitman** [reply]
I would like to participat, yes

**Reply by pixiesue** [reply]
> I would like to participat, yes

https://github.com/orgs/FundamentalFrequency/discussions/6#discussion-4157253

**Reply by stonepreston** [reply]
I think it would be best to order non RGB PCBs first, since I know the current PCB works. Waiting for RGB support could be a while, and even then, I would want to order a small amount of the new design first to make sure everything works first before ordering a large batch.
